pdfgen/src/types/hierarchy/primitives/font.rs

```diff
@@ -0,0 +1,71 @@
+//! Implementation of PDF Font object.
+
+use crate::types::{self, constants};
+
+use super::{name::Name, obj_id::ObjId, object::Object};
+
+/// Represents a font object in a PDF document.
+/// This struct represents a font object in a PDF document, encapsulating the info required to
+/// define and reference a font, including its unique ID, subtype, and base font type.
+/// Fonts are essential for rendering text in PDFs and specify the appearance and
+/// characteristics of text elements.
+pub struct Font {
+    /// ID of this Font object.
+    id: ObjId,
+
+    /// Specifies the subtype of the font, defining its role or characteristics within the PDF.
+    subtype: Vec<u8>,
+
+    /// Represents the base font type, identifying the general font family or format.
+    base_type: Vec<u8>,
+}
+
+impl Font {
+    const FONT: Name<'static> = Name::from_static(b"Font");
+    const SUBTYPE: Name<'static> = Name::from_static(b"Subtype");
+    const BASE_FONT: Name<'static> = Name::from_static(b"BaseFont");
+
+    /// Create a new Font object with the provided id, subtype and base_type.
+    pub fn new(id: ObjId, subtype: Vec<u8>, base_type: Vec<u8>) -> Self {
```

Comment:
This is an example where we can evaluate trade-off between `Name` with generic lifetime and additional (`OwnedName`) type. We could maybe go for a generic `Name` so that `Name<Vec<u8>>` is also a possibility (e.g. `Name<T: AsRef<[u8]>>`). 

That complicates the code a little, but would allow us to have a generic implementation that works for `Name<Vec<u8>>`, `Name<&'a [u8]>`, `Name<String>` etc.

Simply accepting `Vec<u8>` here is not great, because any bytes can be passed here, and semantically we want these bytes to be a valid `Name`.

Reply:
And there's a typo in argument: 

```suggestion
    pub fn new(id: ObjId, subtype: Vec<u8>, base_font: Vec<u8>) -> Self {
```